main.tf

```diff
@@ -1,50 +1,63 @@
 module "s3" {
-  source     = "./modules/s3"
-  model_name = var.model_name
-  tags       = var.tags
-}
+  source = "./modules/s3"
 
+  resource_naming_prefix = var.resource_naming_prefix
+  tags                   = var.tags
+}
 
 module "sagemaker" {
-  source                       = "./modules/sagemaker"
-  tags                         = var.tags
-  sagemaker_execution_role_arn = module.iam.sagemaker_role_arn
-  model_target_variable        = var.model_target_variable
-  algorithm_choice             = var.algorithm_choice
-  tuning_metric                = var.tuning_metric
-  config_bucket_id             = module.s3.config_bucket_id
-  data_location_s3             = var.data_location_s3
-  endpoint_name                = var.endpoint_name
-  model_name                   = var.model_name
-  sagemaker_instance_type      = var.sagemaker_instance_type
-  model_instance_count         = var.model_instance_count
-  ecr_repo_uri                 = module.ecr.ecr_repo_uri
+  source = "./modules/sagemaker"
+
+  # Naming
+  endpoint_name = local.endpoint_name
+  model_name    = local.model_name
+  tags          = var.tags
+
+  # Training
+  algorithm_choice = var.algorithm_choice
+  tuning_metric    = var.tuning_metric
+
+  # Notebook
+  sagemaker_instance_type = var.sagemaker_instance_type
+
+  # Model
+  model_target_variable = var.model_target_variable
+  model_instance_count  = var.model_instance_count
+  ecr_repo_uri          = "${module.ecr.repository.repository_url}:latest"
+
+  # S3
+  config_s3_bucket      = module.s3.config_bucket.id
+  config_bucket_key_arn = module.s3.encryption_key.arn
+  data_s3_bucket        = var.data_s3_bucket
+  data_bucket_key_arn   = var.data_s3_bucket_encryption_key_arn
+  data_location_s3      = var.data_location_s3
+  model_s3_bucket       = module.s3.model_bucket.id
+  model_bucket_key_arn  = module.s3.encryption_key.arn
 }
 
+module "retraining_job" {
+  count  = var.retrain_model_bool ? 1 : 0
+  source = "./modules/glue"
 
-module "iam" {
```

Comment:
I can't see the benefit of separating the Sagemaker IAM role into a different module - I have added the resources into the sagemaker module to stop us from having to pass values between the two.